packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.controller.js

```diff
@@ -0,0 +1,348 @@
+      this.loading.init = true;
+    });
+  }
+
+  getSwitchBtnStatusText(switchBtn) {
+    return this.$translate.instant(
+      `hosting_cdn_shared_state_${switchBtn ? 'enable' : 'disabled'}`,
+    );
+  }
+
+  convertToUnitTime(ttl) {
+    if (ttl % HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY === 0) {
+      return `${ttl /
+        HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY} ${this.$translate.instant(
+        'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_days',
+      )}`;
+    }
+    if (ttl % HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR === 0) {
+      return `${ttl /
+        HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR} ${this.$translate.instant(
+        'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_hours',
+      )}`;
+    }
+    return `${ttl /
+      HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_MINUTE} ${this.$translate.instant(
+      'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_minutes',
+    )}`;
+  }
+
+  getMaxPriority() {
+    return (
+      get(maxBy(this.model.rules, 'config.priority'), 'config.priority') || 0
+    );
+  }
+
+  rulePriorityExist(rule) {
+    return find(
+      this.model.rules,
+      (r) => rule !== r && r.config.priority === rule.config.priority,
+    );
+  }
+
+  setRulesPriority(rule) {
+    if (this.rulePriorityExist(rule)) {
+      forEach(this.model.rules, (iRule) => {
+        const { priority } = iRule.config;
+        if (rule !== iRule && priority >= rule.config.priority) {
+          set(iRule, 'config.priority', priority + 1);
+        }
+      });
+    }
+  }
+
+  markRuleAsToCreat(rule) {
+    this.tasks.toAdd.push(rule);
+  }
+
+  markRuleAsToUpdate(rule) {
+    const i = this.tasks.toUpdate.indexOf(rule);
+    if (i < 0 && rule !== find(this.tasks.toAdd, (r) => r === rule))
+      this.tasks.toUpdate.push(rule);
+  }
+
+  markRuleAsToRemove(rule) {
+    this.tasks.toRemove.push(rule);
+  }
+
+  openCreatCacheRuleModal() {
+    const priority = {
+      max: this.getMaxPriority() + 1,
+      value: this.getMaxPriority() + 1,
+    };
+    const packet = {
+      data: assign({}, this.params, { priority }),
+      rules: this.model.rules,
+      success: (rule) => {
+        this.setRulesPriority(rule);
+        this.markRuleAsToCreat(rule);
+        this.model.rules.push(rule);
+      },
+      fail: (err) => {
+        this.Alerter.error(err.message);
+      },
+      cancel: () => {},
+    };
+    this.$state.go('app.hosting.cdn.shared.addCacheRule', { packet });
+  }
+
+  openUpdateCacheRuleModal(rule) {
+    const priority = {
+      value: rule.config.priority,
+    };
+    const packet = {
+      data: assign({}, this.params, { priority }),
+      rules: this.model.rules,
+      rule,
+      success: (uRule) => {
+        this.setRulesPriority(uRule);
+        this.markRuleAsToUpdate(uRule);
+      },
+      fail: () => {},
+      cancel: () => {},
+    };
+    this.$state.go('app.hosting.cdn.shared.addCacheRule', { packet });
```

Comment:
We use to declare some `bindings/resolve` on components to avoid to inject `$state` in each controller (like `goBack` for example)

Reply:
fixed